pkg/k8s/apis/cilium.io/v2/client/register.go

```diff
@@ -526,990 +418,22 @@ func needsUpdate(clusterCRD *apiextensionsv1beta1.CustomResourceDefinition) bool
-				Type: "array",
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &apiextensionsv1beta1.JSONSchemaProps{
-						Type: "string",
-					},
-				},
-			},
-			"toPorts": {
-				Description: "ToPorts is a list of destination ports identified by port number " +
-					"and protocol which the endpoint subject to the rule is allowed to connect " +
-					"to.\n\nExample: Any endpoint with the label \"role=frontend\" is allowed " +
-					"to initiate connections to destination port 8080/tcp",
-				Type: "array",
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &PortRule,
-				},
-			},
-			"toServices": {
-				Description: "ToServices is a list of services to which the endpoint subject " +
-					"to the rule is allowed to initiate connections.\n\nExample: Any endpoint " +
-					"with the label \"app=backend-app\" is allowed to initiate connections to " +
-					"all cidrs backing the \"external-service\" service",
-				Type: "array",
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &Service,
-				},
-			},
-			"toEndpoints": {
-				Description: "ToEndpoints is a list of endpoints identified by an " +
-					"EndpointSelector to which the endpoint subject to the rule" +
-					"is allowed to communicate.\n\nExample: Any endpoint with the label " +
-					"\"role=frontend\" can be consumed by any endpoint carrying the label " +
-					"\"role=backend\".",
-				Type: "array",
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &EndpointSelector,
-				},
-			},
-			"toRequires": {
-				Description: "ToRequires is a list of additional constraints which must be " +
-					"met in order for the selected endpoints to be able to reach other " +
-					"endpoints. These additional constraints do not by themselves grant access " +
-					"privileges and must always be accompanied with at least one matching " +
-					"FromEndpoints.\n\nExample: Any Endpoint with the label \"team=A\" " +
-					"requires any endpoint to which it communicates to also carry the label " +
-					"\"team=A\".",
-				Type: "array",
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &EndpointSelector,
-				},
-			},
-			"toGroups": {
-				Type: "array",
-				Description: `ToGroups is a list of constraints that will
-				gather data from third-party providers and create a new
-				derived policy.`,
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &apiextensionsv1beta1.JSONSchemaProps{
-						Type: "object",
-						Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
-							"aws": AWSGroup,
-						},
-					},
-				},
-			},
-			"toFQDNs": {
-				Description: `ToFQDNs is a list of rules matching fqdns that endpoint
-				is allowed to communicate with`,
-				Type: "array",
-				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
-					Schema: &FQDNRule,
-				},
-			},
-		},
-	}
+	// TODO(christarazi): Is CNPCRV still needed? It only exists for the
```

Comment:
Yes, it is still needed, it's used to perform the validation in the preflight command